Add more Build service secrets lookup options (#48)
konflux-ci/docs: docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc

This secret will be used by the build service to perform both default and Pipeline-as-Code builds.

It is also possible to have secrets for per-repository or organization access. To do this, a `appstudio.redhat.com/scm.repository` annotation should be added to the secret. It may either specify the full repository path or the

> partial path with a wildcard. For example, to create a secret for all repositories in the `my-user` organization, it should be created as follows:

Reviewer: Lines 71 and 72 should be one line.

> partial path with a wildcard. For example, to create a secret for all repositories in the `my-user` organization, it should be created as follows:

Reviewer: ```suggestion
partial path with a wildcard. For example, to create a secret for all repositories in the `my-user` organization, create (or add) the following YAML file:
```

I mean that we need an imperative-based phrase "to create a secret..., do this..."

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: pipelines-as-code-secret
  namespace: <YOUR NAMESPACE>
  labels:
    appstudio.redhat.com/credentials: scm
    appstudio.redhat.com/scm.host: <gitlab-host> # for example, gitlab.com
  annotations:
    appstudio.redhat.com/scm.repository: my-user/*
type: kubernetes.io/basic-auth
stringData:
  password: <PERSONAL GITLAB ACCESS TOKEN>
----

For a specific repository, the following secret should be created:

[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:
  name: pipelines-as-code-secret
  namespace: <YOUR NAMESPACE>
  labels:
    appstudio.redhat.com/credentials: scm
    appstudio.redhat.com/scm.host: <gitlab-host> # for example, gitlab.com
  annotations:
    appstudio.redhat.com/scm.repository: <repository-path> # for example, my-user/my-repo
type: kubernetes.io/basic-auth
stringData:
  password: <PERSONAL GITLAB ACCESS TOKEN>
----

[NOTE]
====

> It is also allowed to have multiple repositories listed under the `appstudio.redhat.com/scm.repository` annotation (comma separated). The secret will be used for all repositories that match the specified paths.

Reviewer: ```suggestion
You can have multiple repositories listed under the `appstudio.redhat.com/scm.repository` annotation. Separate repository names with commas when listing them. The secret will be used for all repositories that match the specified paths.
```

We try to keep RHTAP/Konflux docs more user-focused and action-based, so it's preferred to say "you can" or "do this".

====

[IMPORTANT]
==== 

> * Secrets lookup mechanism is searching for the most specific secret first. So the secret with repository annotation will be used first if it matches the component repository path. In none found, then lookup will try to find the one with wildcard, or just the host matching one.

Reviewer: ```suggestion
* Secrets lookup mechanism is searching for the most specific secret first. The secret with a repository annotation will be used first if it matches the component repository path. In none found, then a lookup will try to find a secret with a wildcard, or the host matching one.
```
